Python: Add file handling support to BinaryContent for OpenAI Responses API

ltwlf opened the pull request:
## Summary

  Enhances `BinaryContent` to support file handling for OpenAI Responses API, enabling file uploads through the responses agent while maintaining a provider-agnostic design.

  ## Changes

  ### BinaryContent Enhancements
  - **Add `can_read` property**: Indicates whether content has readable data available
  - **Add `from_file()` class method**: Creates BinaryContent instances from file paths with automatic base64 encoding
  - **Fix Unicode handling**: Prevents decode errors when processing binary files (PDFs, images, etc.)

  ### OpenAI Responses Agent Integration
  - **Add BinaryContent support**: Pattern matching case for file handling in `responses_agent_thread_actions.py`
  - **Correct OpenAI API format**: Uses proper `filename` and `file_data` structure with data URI format
  - **UUID-based filenames**: Generates appropriate filenames with mime-type extensions
  - **Provider-specific mapping**: File format conversion happens only in OpenAI agent code

  ### Testing
  - **Test coverage**: New tests for `can_read`, `from_file()`, and binary data handling
  - **Unicode error prevention**: Specific test for binary PDF-like content
  - **Base64 encoding verification**: Ensures proper data format for API compatibility

  ## Design Principles

  - **Provider-agnostic**: BinaryContent remains completely generic with no OpenAI-specific dependencies
  - **Clean separation**: OpenAI format mapping isolated to OpenAI agent files
  - **No FileContent class**: Enhances existing BinaryContent instead of introducing new types
  - **Follows existing patterns**: Similar approach to ImageContent and TextContent handling

  ## Usage Example

  ```python
 response = await self.agent.get_response(
        messages=ChatMessageContent(
            content="Analyse PDF",
            role=AuthorRole("user"),
            items=[BinaryContent.from_file(file_path=c":/test.pdf")],
        )
    )

moonbox3 commented:
Hi @ltwlf, can you please check your Ruff settings? There's a lot of extra line adds and such in this PR, that aren't necessary -- most of the initial changes in I see `responses_agent_thread_actions` for example. It looks like you have some settings overriding our `pyproject.toml` Ruff settings that configure the line-length as 120? 

The current code has the following following 99 chars (allowed because < 120):

```python
from openai.types.responses.response_content_part_added_event import ResponseContentPartAddedEvent
```

and you're trying to commit the change that brings the first import line to 71 chars:

```python
from openai.types.responses.response_content_part_added_event import (
    ResponseContentPartAddedEvent,
)
```

ltwlf commented:
@moonbox3  thanks for the feedback and you are right about the unnecessary formatting changes.

 I tracked down the issue - the root .vscode/settings.json was using autopep8 instead of ruff, which caused imports to be formatted differently than the project's pyproject.toml configuration.

I've fixed the VSCode settings and reverted all the unrelated formatting changes. The PR now only contains the minimal changes needed for FileContent functionality, and future contributions won't have this formatting issue.

eavanvalkenburg reviewed (approved):
Looks great, thanks for the effort!

moonbox3 commented on python/semantic_kernel/agents/open_ai/responses_agent_thread_actions.py:
```diff
@@ -719,6 +721,25 @@ def _prepare_chat_history_for_request(
                             "call_id": content.call_id,
                         }
                         response_inputs.append(rfrc_dict)
+                    case BinaryContent() if content.can_read:
```
Could you please add a sample or two to the [samples/concepts/agents/openai_responses](https://github.com/microsoft/semantic-kernel/tree/main/python/samples/concepts/agents/openai_responses) dir showing how a dev would use/implement this? Once a sample exists, it also needs to be tracked via the [README](https://github.com/microsoft/semantic-kernel/blob/main/python/samples/concepts/README.md). Thanks!

moonbox3 commented on python/samples/concepts/agents/openai_responses/responses_agent_binary_content_upload.py:
```diff
@@ -0,0 +1,159 @@
+
+async def main():
+    # 1. Create the client using OpenAI resources and configuration
+    client, model = OpenAIResponsesAgent.setup_resources()
```
Can we please update to the new pattern of using:

```python
client = OpenAIResponsesAgent.create_client()
```

Then the model is:

```python
from semantic_kernel.connectors.ai.open_ai import OpenAISettings

agent = OpenAIResponsesAgent(
    ai_model_id=OpenAISettings().responses_model_id,
    ...
)
```

moonbox3 commented on python/samples/concepts/agents/openai_responses/responses_agent_binary_content_upload.py:
```diff
@@ -0,0 +1,159 @@
+        )
+
+        # 5. Create initial chat history with uploaded files
+        chat_history = ChatHistory()
```
I'm not sure I understand the use of the `ChatHistory` here. Once the messages are added to it, it's not used for anything else? 

All messages to an agent should be passed in via the `messages` kwarg. They will then be added to the thread.
